evm/vm/base.py

```diff
@@ -188,6 +272,8 @@ def get_block_class(self):
 
     def get_block_by_hash(self, block_hash):
         block_header = self.evm.get_block_header_by_hash(block_hash)
+        if block_header is None:
+            return None
```

Comment:
Trying to understand the `None` return here.  My instinct says this should throw an exception when it can't find the requested block.